src/core.c/Distribution/Path.pm6

```diff
@@ -52,6 +52,7 @@ class Distribution::Path does Distribution::Locally {
     }
 
     method meta(Distribution::Path:D:) { %!meta.item }
+    method modified(--> Instant:D) { $!meta-file.modified }
```

Comment:
We have stopped using file modification times for precomp checks long ago. For one, they don't tell you if a file has actually been modified (could just have been touched or written to with the same content) and there are file systems with a timestamp granularity of 2 whole seconds. Checksums are superior in just about every regard. We should not make a step back here.